Amazon.Ion.ObjectMapper/IonSerializer.cs

```diff
@@ -103,15 +103,37 @@ public object Create(IonSerializationOptions options, IIonReader reader, Type ta
             if (annotations.Length > 0)
             {
                 var typeName = annotations[0];
-                var assemblyName = options.AnnotatedTypeAssemblies.First(a => Type.GetType(FullName(typeName, a)) != null);
-                return Activator.CreateInstance(Type.GetType(FullName(typeName, assemblyName)));
+                Type typeToCreate = null;
+
+                if (options.AnnotatedTypeAssemblies != null)
+                {
+                    foreach (string assemblyName in options.AnnotatedTypeAssemblies)

```

Comment:
Linq can make this better and you could use `Select` to improve readability. Perhaps a future PR to prettify it.

Reply:
Select doesn't help here. 

The more appropriate is FirstOrDefault, but FirstOrDefault returns the string in the annotation list, causing us to call GetType twice which is inefficient.